doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -55,6 +55,7 @@
 - `m_originId` in `mepoo::ChunkHeader` sometimes not set [\#1668](https://github.com/eclipse-iceoryx/iceoryx/issues/1668)
 - Removed `cxx::unique_ptr::reset` [\#1655](https://github.com/eclipse-iceoryx/iceoryx/issues/1655)
 - CI uses outdated clang-format [\#1736](https://github.com/eclipse-iceoryx/iceoryx/issues/1736)
+- Avoid UB when accessing `iox::expected` [\#1750](https://github.com/eclipse-iceoryx/iceoryx/issues/1750)
```

Comment:
@elfenpiff I don't fully get what you mean. The following is definitely UB and avoided with this PR?

```
auto sut = expected<Foo, Bar>::create_error(Bar::VALID_ERROR);
auto myValue = sut->memberVariableOfFoo;
```

Because as you wrote the `ìox::variant` will return a `nullptr` which cause UB in `operator->`